src/test/java/StringManipulationTest.java

```diff
@@ -19,131 +22,206 @@ public void tearDown() {
     }
 
+    // Test case to verify removing nth character in a different string
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'm learning to code!");
+        assertEquals("I'm lea nin  to cod !", manipulatedstring.removeNthCharacter(4, true));
     }
 
+    // Test case for removing nth character with an out-of-bounds value of n
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Test string");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.removeNthCharacter(13, true));
     }
 
+    // Test case to verify removing nth character without maintaining spacing in a
+    // different string
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a test.");
+        assertEquals("Ths s  tst", manipulatedstring.removeNthCharacter(3, false));
     }
 
+    // Test case for removing nth character with an invalid value of n
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Test string");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.removeNthCharacter(0, true));
     }
 
+    // Test case to verify removing nth character in a long string
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I am testing the removeNthCharacter method");
+        assertEquals("I a   e t n   h   e o e t C a a t r m t o ", manipulatedstring.removeNthCharacter(2, true));
+    }
+
+    // Test case for removing nth character from a null string
+    @Test
+    public void testRemoveNthCharacter8() {
+        manipulatedstring.setString(null);
+        assertThrows(NullPointerException.class, () -> manipulatedstring.removeNthCharacter(2, true));
     }
 
+    // DO NOT EDIT
     @Test
     public void testGeSubStrings1() {
+        // Set up the initial string
         manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        String[] sStings = manipulatedstring.getSubStrings(3, 4);
 
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
+    // Test case to verify extracting multiple substrings from a string
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
+        // Set up the initial string
+        manipulatedstring.setString("This is my string");
+        String[] subStrings = manipulatedstring.getSubStrings(1, 4);
+        assertEquals(subStrings.length, 4);
+        assertEquals(subStrings[0], "This");
+        assertEquals(subStrings[1], "is");
+        assertEquals(subStrings[2], "my");
+        assertEquals(subStrings[3], "string");
```

Comment:
technically, the string you expect should be the first argument in the assertEquals method, but it gets the job done. could make bug fixing confusing though.